add_to_comps.rb

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env ruby
+require 'nokogiri'
+# Usage ./add_to_comps.rb comps-file package-name [nonscl]
+#
+# Return codes:
+#
+#   0 - successfully added
+#   1 - package is already present
+
+
+def find_packagereq(comps, package_name)
+  comps.xpath('//packagelist/packagereq').select { |s| s.text == package_name }
+end
+
+# Gets subset of packagereqs that are not -docs nor comments
+def get_packagereqs(packagelist)
+  relevant = []
+  packagelist.children.each do |packagereq|
+    # Break as soon as you find a comment which means start
+    # of nonscl packages or start of -doc packages
+    break if packagereq.class == Nokogiri::XML::Comment
+    relevant << packagereq
+    packagereq.unlink
+  end
+  relevant
+end
+
+def remove_nonscl_comment(packagelist)
+  nonscl_comment = packagelist.children.first
+  nonscl_comment.unlink
+  nonscl_comment
+end
+
+def prepend_packagereqs(packagelist, packagereqs)
+  packagereqs.each do |packagereq|
+    packagelist.children.before(packagereq)
+  end
+end
+
+comps = Nokogiri::XML(File.open(ARGV[0])) { |config| config.noblanks }
+exit(1) unless find_packagereq(comps, ARGV[1]) == []
+packagelist = comps.xpath("//packagelist").first
+new_packagereq = Nokogiri::XML(
+  "<packagereq type='default'>#{ARGV[1]}</packagereq>"
+).children.first
+if ARGV[2] == 'nonscl'
+  scl_packagereqs = get_packagereqs(packagelist)
+  scl_packagereqs.sort! { |x,y| y.text <=> x.text }
```

Comment:
That's fine then, just add another commit re-sorting the files so it doesn't fall to the first person to add a new package.